marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/mapping-functions/core.sjs

```diff
@@ -151,6 +163,19 @@ function parseDateTime(value, pattern) {
   let supportedFormats = ["YYYYMMDDThhmmss", "DD/MM/YYYY-hh:mm:ss", "DD/MM/YYYY hh:mm:ss", "YYYY/MM/DD-hh:mm:ss" , "YYYY/MM/DD hh:mm:ss"];
   let response;
   let errorMsg = `The pattern '${pattern}' cannot be applied to the value '${value}'`;
+
+  if (value) {
```

Comment:
This logic is almost identical. You can reduce the duplication and further remove the need for any comments by introducing a new function here. 

The 'replace(", ", ",");' bit should then be done on "value" after the new function eturns false. 

The code would then read e.g.:

    if (isNullOrEmpty(value)) {
      return null;
    }